frontend/packages/topology/src/actions/TopologyActions.tsx

```diff
@@ -9,16 +9,16 @@ type TopologyActionsProps = {
 };
 
 const TopologyActions: React.FC<TopologyActionsProps> = ({ element }) => {
+  const resource = getResource(element);
```

Comment:
Try using `useDeepCompareMemoize` to avoid calling `useMemo` multiple times.

```suggestion
  const resource = getResource(element);
  const memoizedRes = useDeepCompareMemoize(resource);
```